include/adm/elements/audio_pack_format_hoa.hpp

```diff
@@ -0,0 +1,146 @@
+/// @file audio_pack_format_hoa.hpp
+#pragma once
+
+#include "adm/elements/screen_ref.hpp"
+#include "adm/elements/normalization.hpp"
+#include "adm/elements/nfc_ref_dist.hpp"
+#include "adm/elements/audio_pack_format.hpp"
+
+namespace adm {
+
+    class Document;
+
+    class AudioPackFormatHoa : public AudioPackFormat {
+     public:
+        template <typename... Parameters>
+        static std::shared_ptr<AudioPackFormatHoa> create(
+            AudioPackFormatName name, TypeDescriptor channelType,
```

Comment:
Also a good proposal.

Reply:
This has been addressed by introducing a second constructor for the AudioPackFormat, which is only called from the derived class. If the AudioPackFormat constructor is called directly with type format HOA it will now raise an exception.